core/src/main/java/fr/sncf/osrd/envelope_sim_infra/InfraPathGrade.java

```diff
@@ -0,0 +1,96 @@
+package fr.sncf.osrd.envelope_sim_infra;
+
+import static java.lang.Math.*;
+
+import fr.sncf.osrd.envelope_sim.PhysicsPath;
+import fr.sncf.osrd.train.TrackSectionRange;
+import fr.sncf.osrd.train.TrainPath;
+import fr.sncf.osrd.utils.*;
+import fr.sncf.osrd.utils.graph.EdgeDirection;
+import java.util.*;
+
+public final class InfraPathGrade implements PhysicsPath {
```

Comment:
Codecov is angry, the whole file isn't tested

Reply:
indeed, it's dead code